homeassistant/components/arduino/sensor.py

```diff
@@ -45,9 +57,17 @@ def __init__(self, name, pin, pin_type, board):
         self.direction = "in"
         self._value = None
 
-        board.set_mode(self._pin, self.direction, self.pin_type)
+        board.set_mode(self._pin, self.direction, self.pin_type, self.cb_value_changed)
         self._board = board
 
+        t = Timer(20.0, self.set_initial_value)
```

Comment:
Why should we reset the value after 20 seconds?

We should probably use our event helpers instead, eg `helpers.event.call_later`.

Reply:
The `set_initial_value` only sets the initial value if no value has been received yet. See first if statement in that function.

The reason why I have done this is, because 'PyMata' is used for communication with the Arduino device. In their code if the initial value remains 0, we are not notified of this value. (Personally I think this is a bug) (The development for the version of PyMata we use is stopped and no changes can be made to it)

So if we don't receive a value within the first 20 seconds, I set the value to zero.
We could set the initial value directly to 0, but I thought this could have unwanted side effects for certain automations, because before the first value is received, we could wrongly assume that the value was 0.

For your second remark, I will try to make the changes to use that function.

Reply:
Ok. But why is 0 a more accurate value than `None`? `None` means we don't know what the value is. Do we really know that the initial value is 0? If we don't know that, we should keep `None` as value until the sensor reports its value.

Reply:
I have looked into the code of PyMata (the version that is used in home assistant) and it works as following:
- If the initial value is 0, it is not reported 
  (This is the bug I was talking about, I think this should also be reported)
  (It is because they check if the value is different from the previous value, and the previous 
  value is initialized with 0, which results in the value not being reported to us, because it's the 
  same value).
  Afterwards all the changes are reported.
- If the initial value is different from 0 it is reported and also all the changes are reported

So we can be sure that the value is 0, if we don't receive a value.

Reply:
Ok.